### PR TITLE
Convert OpenACC ParallelReduce

### PR DESCRIPTION
**Pull request by masterleinad**
Part of #5908.

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
What about `OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp`?

**Comment by masterleinad** [comment]
> What about `OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp`?

We are constructing the `FunctorAnalysis` object in that case on the host anyway so there is not much of a motivation to change it apart from consistency with the implementation on other backends that must store the `FunctorAnalysis` object instead of the functor. I would rather revisit this when touching all backends again to allow for templated functors in `parallel_scan`.

**Review by dalg24 — APPROVED** [review]

File: core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_MDRange.hpp [hunk 1]
```diff
@@ -85,16 +72,18 @@ class Kokkos::Impl::ParallelReduce<Functor, Kokkos::MDRangePolicy<Traits...>,
...
-        std::conditional_t<is_reducer_v<ReducerType>, ReducerType,
-                           Sum<ValueType>>(val),
+        Kokkos::Experimental::Impl::FunctorAdapter<FunctorType, Policy>(
+            m_functor_reducer.get_functor()),
+        std::conditional_t<
+            std::is_same_v<FunctorType, typename ReducerType::functor_type>,
+            Sum<ValueType>, typename ReducerType::functor_type>(val),
         m_policy);
 
+    reducer.final(&val);
```
Comment (Rombur) [inline core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_MDRange.hpp:86]: Why do we call `final` now and not before? Was there a bug?

Comment (masterleinad) [reply core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_MDRange.hpp:86]: Yes, this was missing and I tried to enforce it multiple times. I thought this was a good opportunity but, of course, this is somewhat of a drive-by-change.

**Review by seyonglee — APPROVED** [review]
It looks good to me too.

**Comment by dalg24** [comment]
Ignoring HIP and SYCL failures

**Pull request merged**